docs/_docs/reference/experimental/capture-checking/mutability.md

```diff
@@ -75,13 +75,13 @@ an update method or an update class as non-private member or constructor.
 
 When we create an instance of a mutable type we always add `cap` to its capture set. For instance, if class `Ref` is declared as shown previously then `new Ref(1)` has type `Ref[Int]^`.
 
-**Restriction:** A non-mutable type cannot be downcast by a pattern match to a mutable type.
+**Restriction:** A non-mutable type cannot be downcast by a pattern match to a mutable type. (Note: This is currently not enforced)
```

Comment:
I thought a bit about this and was wondering whether we can do this by inspecting capture sets in the pattern: one may never "widen" a readonly set to a exclusive one since that will be a permission upgrade.

So instead of preventing narrowing we prevent widening, very much like what we are already doing for capture sets.